File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.py
def solution(s):
    answer = [0, 0] # 변환 횟수, 제거한 0의 갯수
    
    
    while s != '1':
        before = len(s)
        s = s.replace('0', '')
        after = len(s)
        
        answer[1] += (before-after)
        s = format(after, 'b')
        
        answer[0] += 1
    return answer

File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.java
class Solution {
    public int[] solution(String s) {
        int[] answer = new int[2];
        
        while (!s.equals("1")) {
            int before = s.length();
            s = s.replace("0", "");
            int after = s.length();
            
            // 삭제된 0의 개수 
            answer[1] += (before - after);
                    
            // 이진 변환 결과
            s = Integer.toBinaryString(s.length());
            
            // 이진 변환 횟수
            answer[0]++;
        }
        return answer;
    }
}